src/aliceVision/sfm/pipeline/sequential/ReconstructionEngine_sequentialSfM.cpp

```diff
@@ -187,96 +189,104 @@ void ReconstructionEngine_sequentialSfM::RobustResectionOfImages(
+                                  << " sub-pose and pose defined.");
+#pragma omp critical                              
             set_remainingViewId.erase(possible_resection_index);
+            
             continue;
           }
-
+          
           // We cannot localize a view if it is part of an initialized RIG with unknown Rig Pose
           const bool knownPose = _sfm_data.existsPose(view);
           const Rig& rig = _sfm_data.getRig(view);
           const RigSubPose& subpose = rig.getSubPose(view.getSubPoseId());
-
+          
           if(rig.isInitialized() &&
              !knownPose &&
              (subpose.status == ERigSubPoseStatus::UNINITIALIZED))
           {
             ALICEVISION_LOG_DEBUG("Resection of image " << currentIndex << " ID=" << possible_resection_index << " was skipped." << std::endl
-                              << "RigID=" << view.getRigId() << " Sub-poseID=" << view.getSubPoseId()
-                              << " Rig initialized but unkown pose and sub-pose.");
+                                  << "RigID=" << view.getRigId() << " Sub-poseID=" << view.getSubPoseId()
+                                  << " Rig initialized but unkown pose and sub-pose.");
+#pragma omp critical   
             set_remainingViewId.erase(possible_resection_index);
+            
             continue;
           }
         }
       }
-
-      const bool bResect = Resection(possible_resection_index);
-
-      bImageAdded |= bResect;
-      if (!bResect)
-      {
-        set_rejectedViewId.insert(possible_resection_index);
-        ALICEVISION_LOG_DEBUG("Resection of image " << currentIndex << " ID=" << possible_resection_index << " was not possible.");
-      }
-      else
-      {
-        set_reconstructedViewId.insert(possible_resection_index);
-        ALICEVISION_LOG_DEBUG("Resection of image: " << currentIndex << " ID=" << possible_resection_index << " succeed.");
-        _sfm_data.GetViews().at(possible_resection_index)->setResectionId(resectionId);
-        ++resectionId;
-      }
-      set_remainingViewId.erase(possible_resection_index);
-
-      // Limit to a maximum number of cameras added to ensure that
-      // we don't add too much data in one step without bundle adjustment.
-      if(_sfm_data.getValidViews().size() - prevReconstructedViews.size() > maxImagesPerGroup)
+      
+      ResectionData newResectionData;
+      bool bResect = computeResection(possible_resection_index, newResectionData);
+#pragma omp critical      
       {
-        break;
+        if (bResect)
+        {
+          bImageAdded |= bResect;
+          updateScene(possible_resection_index, newResectionData);
+          set_reconstructedViewId.insert(possible_resection_index);
+          ALICEVISION_LOG_DEBUG("Resection of image: " << currentIndex << " ID=" << possible_resection_index << " succeed.");
+          _sfm_data.GetViews().at(possible_resection_index)->setResectionId(resectionId);
```

Comment:
As there is no more link between resections of the same resectionGroup, we can remove `resectionId` and do:
`_sfm_data.GetViews().at(possible_resection_index)->setResectionId(resectionGroupIndex);`